baremetal/metal3data_manager_test.go

```diff
@@ -1319,14 +1320,21 @@ var _ = Describe("Metal3Data manager", func() {
 			}
 			if tc.ipClaim != nil {
 				capm3IPClaim := &ipamv1.IPClaim{}
-				poolNamespacedName := types.NamespacedName{
-					Name:      tc.m3d.Name,
-					Namespace: tc.m3d.Namespace,
+				claimNamespacedName := types.NamespacedName{
+					Name:      tc.ipClaim.Name,
+					Namespace: tc.ipClaim.Namespace,
 				}
 
-				err = dataMgr.client.Get(context.TODO(), poolNamespacedName, capm3IPClaim)
-				Expect(err).To(HaveOccurred())
-				Expect(apierrors.IsNotFound(err)).To(BeTrue())
+				err = dataMgr.client.Get(context.TODO(), claimNamespacedName, capm3IPClaim)
+				if tc.injectDeleteErr {
+					// There was an error deleting the claim, so we expect it to still be there
```

Comment:
is this comment aligned with the code? A bit confused here

Reply:
I mean the next line expects error to be nil

Reply:
Ah hmm maybe a bit vague. I meant that we expect an error when deleting, meaning that the IPClaim is still expected to exist (since we could not delete it)

Reply:
We expect error when deleting, which means the IPClaim should still exist. So when we try to `get` it above, we expect no problems, i.e. the error from get should be nil.